site/app/views/admin/UsersView.php

```diff
@@ -295,6 +297,133 @@ public function userForm($reg_sections, $rot_sections, $action, $use_database=fa
+     * @param array  $reg_sections associative array representing registration sections in the system
+     * @param bool   $use_database
+     * @return string
+     */
+    public function copyStudentEmailForm($students,$reg_sections,$use_database=false) {
+        $reg_select_html = "";
+        foreach ($reg_sections as $section) {
+            $section = $section['sections_registration_id'];
+            $reg_select_html .= "<option value='{$section}'>Section {$section}</option>\n";
+        }
+        $emails=array();
+        $emails_corresponding_section=array();
+        foreach($students as $student){
+            array_push($emails,$student->getEmail());
+            if($student->getRegistrationSection()===null){
+                array_push($emails_corresponding_section,"NULL");
+            }
+            else{
+                array_push($emails_corresponding_section,$student->getRegistrationSection());
+            }    
+        }
+        $emails_str=implode(",", $emails);
+        $emails_corresponding_section_str= implode(",",$emails_corresponding_section);
+        $return = <<<HTML
+<div class="popup-form" id="copy-student-email-form">                                                                   
+<form>
+    <div>
+        <input type="radio" name="email_group" id="radio1" value="1"/>
+        Registered Section:
+        <select name="registered_section">
+            <option value="NULL">Not Registered</option>
+            {$reg_select_html}
+        </select>
+    </div><br />
+    <div>
+        <input type="radio" name="email_group" id="radio2" value="2"/>
+        All Registered Students (Non Null)
+    </div><br />
+    <div>
+        <input type="radio" name="email_group" id="radio3" value="3"/>
+        All Users
+    </div><br />
+HTML;
+        if ($use_database) {
+            $return .= <<<HTML
+    <div style="width: 60%">
+        Password:<br />
+        <input type="password" name="user_password" placeholder="New Password" />    
+    </div>
+HTML;
+        }
+
+        $return .= <<<HTML
+    <label for="copymsg" id="copymsgid" style="visibility: hidden; float: right; width: auto;">mails copied!</label><br />    
+    <div style="float: right; width: auto; margin-top: 10px">
+        <a onclick="$('#copy-student-email-form').css('display', 'none');" class="btn btn-danger">Cancel</a>
+        <input class="btn btn-primary" type="button" value="Download" onclick="copyStudentEmail('download','{$emails_str}','{$emails_corresponding_section_str}');" />
+        <input class="btn btn-primary" type="button" value="Copy" onclick="copyStudentEmail('copy','{$emails_str}','{$emails_corresponding_section_str}');" />
+    </div>
+</form>
+</div>
+HTML;
+        return $return;
+    }
+
+
+/**
+     * Creates the form box to be displayed when copying or downloading emails on the graders pages
+     * @param User[] $graders
+     * @param bool   $use_database
+     * @return string
+     */
```

Comment:
this function consist of html code for copy/download form that will be rendered for graders tab